src/containers/my-resources/add-categories-modal/AddCategoriesModal.tsx

```diff
@@ -0,0 +1,87 @@
+import { FC, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import Typography from '@mui/material/Typography'
+import Box from '@mui/material/Box'
+
+import AppTextField from '~/components/app-text-field/AppTextField'
+import AppButton from '~/components/app-button/AppButton'
+import useForm from '~/hooks/use-form'
+
+import {
+  initialValues,
+  validations
+} from '~/containers/my-resources/add-categories-modal/AddCategoriesModal.constants'
+import { styles } from '~/containers/my-resources/add-categories-modal/AddCategories.styles'
+import {
+  ButtonTypeEnum,
+  ButtonVariantEnum,
+  ComponentEnum,
+  CreateCategoriesParams
+} from '~/types'
+
+interface AddCategoriesModalProps {
+  closeModal: () => void
+  createCategories: (params?: CreateCategoriesParams) => Promise<void>
+}
+
+const AddCategoriesModal: FC<AddCategoriesModalProps> = ({
+  closeModal,
+  createCategories
+}) => {
+  const { t } = useTranslation()
+  const [loading, setLoading] = useState<boolean>(false)
+
+  const { data, errors, handleInputChange, handleBlur, handleSubmit } =
+    useForm<CreateCategoriesParams>({
+      initialValues: initialValues,
+      validations,
+      onSubmit: async () => {
+        setLoading(true)
+        await createCategories({
+          name: data.name
```

Comment:
you can simpy pass data to createCategories:
createCategories(data)

Reply:
> you can simpy pass data to createCategories: createCategories(data)

oh, you are right, that's better